### PR TITLE
chore(Java): add isomorphic strings

### PR DESCRIPTION
**Pull request by Ishantgarg-web**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Looks great! Also please update the respective Readme file. In your case, it will be
./algorithms/Java/README.md

**Comment by Ishantgarg-web** [comment]
I updated algorithms/Java/README.md file

File: algorithms/Java/README.md [hunk 1]
```diff
@@ -88,6 +88,7 @@
 - [Boyer Moore Search](strings/Boyer_Moore.java)
 - [Reverse String](strings/reverse-string.java)
 - [First Non Repeating Character](strings/first-non-repeating-char.java)
+- [isomprphic Strings](strings/isomorphic_strings.java)
```
Comment (mohitchakraverty) [inline algorithms/Java/README.md:91]: Please correct the spelling of isomorphic and **I** should be capital.
```suggestion
- [Isomorphic Strings](strings/isomorphic_strings.java)
```

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Also please make the following changes.

File: algorithms/Java/README.md (hunk 1 above)
Comment (Ishantgarg-web) [reply algorithms/Java/README.md]: Changes were done!

**Review by mohitchakraverty — APPROVED** [review]

**Pull request merged**